apps/OpenSpace/ext/launcher/src/launcherwindow.cpp

```diff
@@ -637,20 +678,117 @@ void LauncherWindow::openProfileEditor(const std::string& profile, bool isUserPr
     }
 }
 
-void LauncherWindow::openWindowEditor() {
-    SgctEdit editor(this, _userConfigPath);
-    int ret = editor.exec();
-    if (ret == QDialog::DialogCode::Accepted) {
-        sgct::config::Cluster cluster = editor.cluster();
+void LauncherWindow::editRefusalDialog(const std::string title, const std::string msg) {
+    QMessageBox msgBox;
+    msgBox.setText(QString::fromUtf8(msg.c_str()));
+    msgBox.setWindowTitle(QString::fromUtf8(title.c_str()));
+    msgBox.setIcon(QMessageBox::Warning);
+    msgBox.exec();
+}
 
-        std::filesystem::path savePath = editor.saveFilename();
-        saveWindowConfig(this, savePath, cluster);
-        // Truncate path to convert this back to path relative to _userConfigPath
-        std::string p = savePath.string().substr(_userConfigPath.size());
-        populateWindowConfigsList(p);
+void LauncherWindow::openWindowEditor(const std::string& winCfg, bool isUserWinCfg) {
+    std::string saveWindowCfgPath = isUserWinCfg ? _userConfigPath : _configPath;
+    int ret = QDialog::DialogCode::Rejected;
+    sgct::config::Cluster preview;
+    if (winCfg.empty()) {
+        SgctEdit editor(this, _userConfigPath);
+        ret = editor.exec();
+        if (ret == QDialog::DialogCode::Accepted) {
+            handleReturnFromWindowEditor(
+                editor.cluster(),
+                editor.saveFilename(),
+                saveWindowCfgPath
+            );
+        }
+    }
+    else {
+        try {
+            sgct::config::GeneratorVersion previewGenVersion =
+                sgct::readConfigGenerator(winCfg);
+            sgct::loadFileAndSchemaThenValidate(
+                winCfg,
+                _configPath + "/schema/sgct.schema.json",
+                "This configuration file is unable to generate a proper display"
+            );
+            sgct::loadFileAndSchemaThenValidate(
+                winCfg,
+                _configPath + "/schema/sgcteditor.schema.json",
+                "This configuration file is valid for generating a display, but "
+                "its format does not match the window editor requirements and "
+                "cannot be opened in the editor"
+            );
+            if (previewGenVersion.versionCheck(minimumVersion)) {
+                try {
+                    preview = sgct::readConfig(
+                        winCfg,
+                        "This configuration file is unable to generate a proper display "
+                        "due to a problem detected in the readConfig function"
+                    );
+                }
+                catch (std::runtime_error& e) {
+                    //Re-throw an SGCT error exception with the runtime exception message
+                    throw sgct::Error(
```

Comment:
I don't think we want to throw `sgct::Error`s from outside the SGCT application.  Would be better to throw some other type of error.   The error codes are not meant to be user-extendable